packages/dd-trace/src/encode/chunk.js

```diff
@@ -20,16 +20,13 @@ class Chunk {
 
     if (length < 0x20) { // fixstr
       this.reserve(length + 1)
+      this.buffer[this.length] = length | 0xa0
       this.length += 1
-      this.buffer[offset] = length | 0xa0
     } else if (length < 0x100000000) { // str 32
       this.reserve(length + 5)
+      this.buffer[this.length] = 0xdb
+      this.buffer.writeUInt32BE(length, this.length + 1)
```

Comment:
Why this change? Writing the value manually has been much faster in all of my tests.

Reply:
less GC pressure generated doing it this way, this shouldn't be a significant CPU savings to manually place in 4 bytes, if we really want to bench this, using a DataView would be the fastest these days most likely but adds some GC pressure.

Reply:
It looks like Node does the [same thing](https://github.com/nodejs/node/blob/14aed60941da978346de286b3d5b814327944d38/lib/internal/buffer.js#L800) internally but with additional checks. What makes their version generate less garbage?

Reply:
Odd, lets just skip this for now and I can see what is going on in Node core. I know at least checking via devtools I was seeing some stuff but this seems more like it should be done outside this PR then.